components/automate-cli/pkg/verifyserver/server/api/v1/stopmockserver_test.go

```diff
@@ -0,0 +1,177 @@
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/server"
+	v1 "github.com/chef/automate/components/automate-cli/pkg/verifyserver/server/api/v1"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/startmockserverservice"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/stopmockserverservice"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/gofiber/fiber"
+	"github.com/stretchr/testify/assert"
+)
+
+func SetupMockServices(protocol string) (stopmockserverservice.IStopMockServerService, startmockserverservice.IStartMockServersService) {
+	stopmock := &stopmockserverservice.MockStopMockServerService{
+		StopMockServerFunc: func(server *models.Server) error {
+			if protocol == constants.TCP {
+				return errors.New("Error while shutting down the server")
+			}
+			return nil
+		},
+	}
+
+	servers := []*models.Server{
+		{
+			Port:     5431,
+			Protocol: constants.UDP,
+		},
+		{
+			Port:     1234,
+			Protocol: constants.TCP,
+		},
+	}
+	startmock := &startmockserverservice.MockServersService{
+		GetMockServersFunc: func() []*models.Server {
+			if protocol == constants.HTTPS {
+				return []*models.Server{}
+			}
+			return servers
+		},
+		SetMockServersFunc: func(s []*models.Server) {
+			servers = s
+		},
+	}
+	return stopmock, startmock
+}
+
+func SetupStopMockServerHandlers(sm stopmockserverservice.IStopMockServerService, ss startmockserverservice.IStartMockServersService) (*fiber.App, error) {
+	log, err := logger.NewLogger("text", "debug")
+	if err != nil {
+		return nil, err
+	}
+	fconf := &fiber.Settings{
+		ServerHeader: server.SERVICE,
+		ErrorHandler: fiberutils.CustomErrorHandler,
+	}
+	app := fiber.New(fconf)
+	handler := v1.NewHandler(log).
+		AddMockServerServices(ss).
+		AddStopMockServerService(sm)
+	vs := &server.VerifyServer{
+		Port:    server.DEFAULT_PORT,
+		Log:     log,
+		App:     app,
+		Handler: handler,
+	}
+	vs.SetupRoutes()
+	return vs.App, nil
+}
+
+func TestStopMockServerAPI(t *testing.T) {
+	tests := []struct {
+		description  string
+		expectedCode int
+		expectedBody string
+		reqBody      string
+		protocol     string
+	}{
+		{
+			description:  "200:success status route",
+			expectedCode: 200,
+			expectedBody: "{\"status\":\"SUCCESS\",\"result\":\"Server stop successfully\"}",
+			protocol:     constants.UDP,
+			reqBody: `{
+				"port": 5431,
+				"protocol": "udp"
+			  }`,
+		},
+		{
+			description:  "400:bad request",
+			expectedCode: 400,
+			expectedBody: "{\"status\":\"FAILED\",\"result\":null,\"error\":{\"code\":400,\"message\":\"Stop mock server request body parsing failed: invalid character '\\\\n' in string literal\"}}",
```

Comment:
same

Reply:
these are test cases and used on once so we can use them directly.